src/Widgets/BridgeWidgetApi.ts

```diff
@@ -1,69 +1,211 @@
+                break;
+            default:
+                throw new ApiError("Not a known service, or service doesn't expose a config", ErrCode.NotFound);
+        }
+
+        if (!config) {
+            throw new ApiError("Service is not enabled", ErrCode.DisabledFeature);
+        }
+
+        res.send({
+            config
+        });
+    }
+
+    private async getConfigSection(req: ProvisioningRequest, res: Response<WidgetConfigurationSection[]>) {
+        if (req.params.section !== "general") {
+            throw new ApiError("Not a known config section", ErrCode.NotFound);
+        }
+        res.send(
+            [{
+
+                name: 'Overview',
+                options: [{
+                    key: 'name',
+                    type: WidgetConfigurationType.String,
+                    currentValue: null,
+                    defaultValue: 'Agent Smith',
+                }]
+            }]
+        );
+    }
+
+    private async getConnections(req: ProvisioningRequest, res: Response<GetConnectionsResponseItem[]>) {
+        if (!req.userId) {
+            throw Error('Cannot get connections without a valid userId');
+        }
+        await assertUserPermissionsInRoom(req.userId, req.params.roomId as string, "read", this.intent);
+        const connections = this.connMan.getAllConnectionsForRoom(req.params.roomId as string);
+        const powerlevel = new PowerLevelsEvent({content: await this.intent.underlyingClient.getRoomStateEvent(req.params.roomId, "m.room.power_levels", "")});
+        const details = connections.map(c => c.getProvisionerDetails?.(true)).filter(c => !!c) as GetConnectionsResponseItem[];
+        const userPl = powerlevel.content.users?.[req.userId] || powerlevel.defaultUserLevel;
+
+        for (const c of details) {
+            const requiredPl = Math.max(powerlevel.content.events?.[c.type] || 0, powerlevel.defaultStateEventLevel);
+            c.canEdit = userPl >= requiredPl;
+            if (!c.canEdit) {
+                delete c.secrets;
+            }
+        }
+
+        res.send(details);
     }
 
+    private async getConnectionsForService(req: ProvisioningRequest, res: Response<GetConnectionsForServiceResponse<GetConnectionsResponseItem>>) {
+        if (!req.userId) {
+            throw Error('Cannot get connections without a valid userId');
+        }
+        await assertUserPermissionsInRoom(req.userId, req.params.roomId as string, "read", this.intent);
+        const allConnections = this.connMan.getAllConnectionsForRoom(req.params.roomId as string);
+        const powerlevel = new PowerLevelsEvent({content: await this.intent.underlyingClient.getRoomStateEvent(req.params.roomId, "m.room.power_levels", "")});
+        const connections = allConnections.map(c => c.getProvisionerDetails?.(true)).filter(c => c?.service === req.params.service).filter(c => !!c) as GetConnectionsResponseItem[];
+        const userPl = powerlevel.content.users?.[req.userId] || powerlevel.defaultUserLevel;
```

Comment:
The code here looks largely the same as the one in getConnections(), perhaps some refactoring is in order?

Reply:
Done!